### ISSUE TITLE
Encoding failure of column names in Windows 10 (but not Linux)

### ISSUE THREAD

**Issue by mathesong** [r-dbi/RSQLite#284]
I apologise if this problem is considered solved, but I cannot seem to work it out from any of the previous issues.

## The problem

I have column names with Swedish characters, and they don't seem to work when going into or out of my database.  Frustratingly, they're all rendered by exactly the same set of characters, so I can't fix them post hoc either.  I'll provide the reprexes below.

## Windows
``` r
library(DBI)
library(RSQLite)
#> Warning: package 'RSQLite' was built under R version 3.5.3
library(tidyverse)
#> Warning: package 'ggplot2' was built under R version 3.5.3
#> Warning: package 'tibble' was built under R version 3.5.3
#> Warning: package 'purrr' was built under R version 3.5.3

data(mtcars)

normaltable <- mtcars
enctable <- mtcars %>% 
  rename("mpög" = mpg, "cåyl" = cyl, "däsp" = disp)


mydb <- dbConnect(RSQLite::SQLite(), 
                  "testdb.sqlite")

dbWriteTable(mydb, "normaltable", normaltable[1:5,])
dbReadTable(mydb, "normaltable")
#>    mpg cyl disp  hp drat    wt  qsec vs am gear carb
#> 1 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4
#> 2 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4
#> 3 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1
#> 4 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1
#> 5 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2


dbWriteTable(mydb, "enctable", enctable[1:5,], check.names=FALSE)
dbReadTable(mydb, "enctable")
#>   mpÃ.g cÃ.yl dÃ.sp  hp drat    wt  qsec vs am gear carb
#> 1  21.0     6   160 110 3.90 2.620 16.46  0  1    4    4
#> 2  21.0     6   160 110 3.90 2.875 17.02  0  1    4    4
#> 3  22.8     4   108  93 3.85 2.320 18.61  1  1    4    1
#> 4  21.4     6   258 110 3.08 3.215 19.44  1  0    3    1
#> 5  18.7     8   360 175 3.15 3.440 17.02  0  0    3    2


dbDisconnect(mydb)

file.remove("testdb.sqlite")
#> [1] TRUE

Sys.getlocale()
#> [1] "LC_COLLATE=English_United Kingdom.1252;LC_CTYPE=English_United Kingdom.1252;LC_MONETARY=English_United Kingdom.1252;LC_NUMERIC=C;LC_TIME=English_United Kingdom.1252"
```

<sup>Created on 2019-04-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

And here is my session info:

```
R version 3.5.2 (2018-12-20)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] shiny_1.2.0          forcats_0.4.0        stringr_1.4.0        dplyr_0.8.0.1        purrr_0.3.2         
 [6] readr_1.3.1          tidyr_0.8.3          tibble_2.1.1         ggplot2_3.1.1        tidyverse_1.2.1     
[11] RSQLite_2.1.1        analogsea_0.6.6.9313 RMySQL_0.10.17       DBI_1.0.0           

loaded via a namespace (and not attached):
 [1] httr_1.4.0       bit64_0.9-7      jsonlite_1.6     modelr_0.1.4     assertthat_0.2.1 askpass_1.1     
 [7] blob_1.1.1       cellranger_1.1.0 yaml_2.2.0       pillar_1.3.1     backports_1.1.3  lattice_0.20-38 
[13] glue_1.3.1       digest_0.6.18    promises_1.0.1   rvest_0.3.2      colorspace_1.4-1 htmltools_0.3.6 
[19] httpuv_1.4.5.1   plyr_1.8.4       clipr_0.5.0      pkgconfig_2.0.2  broom_0.5.1      haven_2.1.0     
[25] xtable_1.8-3     scales_1.0.0     processx_3.3.0   whisker_0.3-2    later_0.8.0      openssl_1.3     
[31] generics_0.0.2   ssh_0.5          withr_2.1.2      credentials_1.1  lazyeval_0.2.2   cli_1.1.0       
[37] magrittr_1.5     crayon_1.3.4     readxl_1.3.1     mime_0.6         memoise_1.1.0    evaluate_0.13   
[43] ps_1.3.0         fs_1.2.6         nlme_3.1-137     xml2_1.2.0       tools_3.5.2      hms_0.4.2       
[49] munsell_0.5.0    reprex_0.2.1     callr_3.1.1      compiler_3.5.2   rlang_0.3.4      grid_3.5.2      
[55] rstudioapi_0.9.0 sys_3.1          miniUI_0.1.1.1   rmarkdown_1.12   gtable_0.3.0     curl_3.3        
[61] R6_2.4.0         lubridate_1.7.4  knitr_1.22       bit_1.1-14       stringi_1.4.3    Rcpp_1.0.1      
[67] tidyselect_0.2.5 xfun_0.6  
```

## Ubuntu

``` r
library(DBI)
library(RSQLite)
library(tidyverse)

data(mtcars)

normaltable <- mtcars
enctable <- mtcars %>% 
  rename("mpög" = mpg, "cåyl" = cyl, "däsp" = disp)


mydb <- dbConnect(RSQLite::SQLite(), 
                  "testdb.sqlite")

dbWriteTable(mydb, "normaltable", normaltable[1:5,])
dbReadTable(mydb, "normaltable")
#>    mpg cyl disp  hp drat    wt  qsec vs am gear carb
#> 1 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4
#> 2 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4
#> 3 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1
#> 4 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1
#> 5 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2


dbWriteTable(mydb, "enctable", enctable[1:5,], check.names=FALSE)
dbReadTable(mydb, "enctable")
#>   mpög cåyl däsp  hp drat    wt  qsec vs am gear carb
#> 1 21.0    6  160 110 3.90 2.620 16.46  0  1    4    4
#> 2 21.0    6  160 110 3.90 2.875 17.02  0  1    4    4
#> 3 22.8    4  108  93 3.85 2.320 18.61  1  1    4    1
#> 4 21.4    6  258 110 3.08 3.215 19.44  1  0    3    1
#> 5 18.7    8  360 175 3.15 3.440 17.02  0  0    3    2


dbDisconnect(mydb)

file.remove("testdb.sqlite")
#> [1] TRUE

Sys.getlocale()
#> [1] "LC_CTYPE=en_US.UTF-8;LC_NUMERIC=C;LC_TIME=en_US.UTF-8;LC_COLLATE=en_US.UTF-8;LC_MONETARY=en_US.UTF-8;LC_MESSAGES=C;LC_PAPER=en_US.UTF-8;LC_NAME=C;LC_ADDRESS=C;LC_TELEPHONE=C;LC_MEASUREMENT=en_US.UTF-8;LC_IDENTIFICATION=C"
```

<sup>Created on 2019-04-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

## System encoding

I also tried adding an enconding="windows-1252" input argument in the original dbConnect() call, and it didn't seem to change anything.  Reprex below:

``` r
library(DBI)
library(RSQLite)
#> Warning: package 'RSQLite' was built under R version 3.5.3
library(tidyverse)
#> Warning: package 'ggplot2' was built under R version 3.5.3
#> Warning: package 'tibble' was built under R version 3.5.3
#> Warning: package 'purrr' was built under R version 3.5.3

data(mtcars)

normaltable <- mtcars
enctable <- mtcars %>% 
  rename("mpög" = mpg, "cåyl" = cyl, "däsp" = disp)


mydb <- dbConnect(RSQLite::SQLite(), 
                  "testdb.sqlite", encoding="windows-1251")

dbWriteTable(mydb, "normaltable", normaltable[1:5,])
dbReadTable(mydb, "normaltable")
#>    mpg cyl disp  hp drat    wt  qsec vs am gear carb
#> 1 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4
#> 2 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4
#> 3 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1
#> 4 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1
#> 5 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2



dbWriteTable(mydb, "enctable", enctable[1:5,], check.names=FALSE)
dbReadTable(mydb, "enctable")
#>   mpÃ.g cÃ.yl dÃ.sp  hp drat    wt  qsec vs am gear carb
#> 1  21.0     6   160 110 3.90 2.620 16.46  0  1    4    4
#> 2  21.0     6   160 110 3.90 2.875 17.02  0  1    4    4
#> 3  22.8     4   108  93 3.85 2.320 18.61  1  1    4    1
#> 4  21.4     6   258 110 3.08 3.215 19.44  1  0    3    1
#> 5  18.7     8   360 175 3.15 3.440 17.02  0  0    3    2



dbDisconnect(mydb)

file.remove("testdb.sqlite")
#> [1] TRUE

Sys.getlocale()
#> [1] "LC_COLLATE=English_United Kingdom.1252;LC_CTYPE=English_United Kingdom.1252;LC_MONETARY=English_United Kingdom.1252;LC_NUMERIC=C;LC_TIME=English_United Kingdom.1252"
```

<sup>Created on 2019-04-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

If anyone can help, I would be extremely thankful. I'm hopefully not the only one struggling with this.

Thanks so much in advance!

**Comment by krlmlr** [comment]
Thanks. This appears to be a bug, but I don't have a Windows system to test this right now.

Can you please double-check (e.g. with sqlitebrowser) if the column names are stored correctly in the database, or garbled?

**Comment by mathesong** [comment]
Thanks so much for the help! 

I just downloaded sqlitebrowser, opened it up, and I see that the column names are in fact stored _correctly_ within the database.  So it appears that it's only when I extract them that they become garbled.

**Comment by krlmlr** [comment]
Thanks again. I can't reproduce using the development version, looks like this has been fixed in #276. Can you please install from source:

```r
devtools::install_github("r-dbi/RSQLite")
```

You'll need Rtools for that.

**Comment by mathesong** [comment]
O, amazing - that solves it! I should really have downloaded the dev version in advance - sorry about that! Thanks so much for the help!!
